cmd/snap-preseed/preseed_linux.go

```diff
@@ -215,14 +246,6 @@ func prepareChroot(preseedChroot string) (func(), error) {
 		}
 	}
 
-	// read version from the mounted core snap
-	infoPath := filepath.Join(snapdMountPath, dirs.CoreLibExecDir, "info")
```

Comment:
with this change we fail later now, that seems a bit annoying.  can't we pass the directory as seen from outside to choose... ? then choose return a struct and we pass the struct to runPreseed ?

Reply:
yet, another approach  is to leave Check... as it is,  and simply do the second comparision only between snap and deb in runPreseed. If the deb is older than the snap we don't care either way.

Reply:
I liked the first idea more, did that (except for passing the directory.. bit), I hope it makes sense?